modules/engine/src/model/model.ts

```diff
@@ -565,23 +578,25 @@ export class Model {
 
   _updatePipeline(): RenderPipeline {
     if (this._pipelineNeedsUpdate) {
+      const prevShaders = [] as Shader[];
       if (this.pipeline) {
         log.log(
           1,
           `Model ${this.id}: Recreating pipeline because "${this._pipelineNeedsUpdate}".`
         )();
+        prevShaders.push(this.pipeline.vs, this.pipeline.fs);
       }
 
       this._pipelineNeedsUpdate = false;
 
-      const vs = this.device.createShader({
+      const vs = this.shaderFactory.createShader({
         id: `${this.id}-vertex`,
```

Comment:
How do we handle the id in the shader cache hash? 
- If we ignore it, the first id would be frozen, so we might as well overwrite it or decorate it somehow (`-cached`?) to make clear that it might not be correct.
- If we include it, we could create multiple identical shaders with just different ids, but ids would be correct.
I think this behavior should be documented.

Reply:
The `-cached` suffix feels reasonable to me – I've gone with that for now, and will include it in the docs.